sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py

```diff
@@ -530,3 +536,187 @@ async def analyze_sentiment(  # type: ignore
+            take precedence over whole batch language. See https://aka.ms/talangs for
+            supported languages in Text Analytics API.
+        :keyword bool show_stats: If set to true, response will contain document level statistics.
+        :keyword int polling_interval: Waiting time between two polls for LRO operations
+            if no Retry-After header is present. Defaults to 30 seconds.
+        :keyword str continuation_token: A continuation token to restart a poller from a saved state.
+        :return: An instance of an AsyncLROPoller. Call `result()` on the poller
+            object to return an instance of TextAnalysisResult.
+        :raises ~azure.core.exceptions.HttpResponseError or TypeError or ValueError or NotImplementedError:
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_analyze_text.py
+                :start-after: [START analyze_text]
+                :end-before: [END analyze_text]
+                :language: python
+                :dedent: 8
+                :caption: Start a long-running operation to perform a variety of text analysis tasks over a batch of documents.
+        """
+
+        display_name = kwargs.pop("display_name", None)
+        language_arg = kwargs.pop("language", None)
+        language = language_arg if language_arg is not None else self._default_language
+        docs = self._client.models(api_version="v3.1-preview.3").MultiLanguageBatchInput(documents=_validate_input(documents, "language", language))
+        show_stats = kwargs.pop("show_stats", False)
+        polling_interval = kwargs.pop("polling_interval", self._client._config.polling_interval)
+        continuation_token = kwargs.pop("continuation_token", None)
+
+        try:
+            analyze_tasks = self._client.models(api_version='v3.1-preview.3').JobManifestTasks(
+                entity_recognition_tasks = [t.to_generated() for t in entities_recognition_tasks],
+                entity_recognition_pii_tasks = [t.to_generated() for t in pii_entities_recognition_tasks],
+                key_phrase_extraction_tasks = [t.to_generated() for t in key_phrase_extraction_tasks]
+                # TODO: add custom task types later
+            )
+            analyze_body = self._client.models(api_version='v3.1-preview.3').AnalyzeBatchInput(
+                display_name=display_name,
+                tasks=analyze_tasks,
+                analysis_input=docs
+            )
+            return await self._client.begin_analyze(
+                body=analyze_body,
+                cls=kwargs.pop("cls", self._analyze_result_callback),
+                polling=AsyncLROBasePolling(timeout=polling_interval, **kwargs),
+                **kwargs
+            )
+
+        except NameError:
+            raise NotImplementedError("Service method 'begin_analyze_text' is only available for API versions v3.2-preview.1 and up.")
```

Comment:
to be consistent, this should raise a `ValueError`, I also don't think catching `NameError` would catch this, see how we do it in [pii](https://github.com/Azure/azure-sdk-for-python/blob/65f1a664a7f252cf71b14160a6a680d890c5ec1c/sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py#L311) for example.

P.S., the final `ValueError` returned to the customer should refer to the method as `begin_analyze` since that's what they see it as

Reply:
I updated this here and in all other relevant places too (for Healthcare and Analyze in both sync and async clients).